shell/browser/api/electron_api_web_contents_mac.mm

```diff
@@ -11,6 +11,13 @@
 
 namespace api {
 
+void WebContents::Focus() {
+  if (GetType() != Type::BACKGROUND_PAGE) {
```

Comment:
We should avoid testing type as new types may be added in future and things could break.

You can just test `owner_window()` here and there is no need for Cocoa code. And there should be a comment saying why this is needed:


```c++
void WebContents::Focus() {
  // Focusing on WebContents does not automatically focus the window on macOS,
  // do it manually to match the behavior on Windows.
#if defined(OS_MACOSX)
  if (owner_window())
    owner_window()->Focus(true);
#endif
  web_contents()->Focus();
}
```